meta-mentor-staging/recipes-core/udev/udev-extraconf/mount_modified.sh

```diff
@@ -19,10 +19,17 @@ done
 
 automount() {	
 	name="`basename "$DEVNAME"`"
-        if [ -z ${ID_FS_LABEL} ]; then
-		#Lets use UUID as the LABEL, as it will be unique for the $DEVNAME
-		ID_FS_LABEL=`/sbin/blkid | grep $DEVNAME | awk {'print $2'} | tr '=|"' ' ' | awk {'print $2'}`
-        fi
+
+	# If the $DEVNAME has a LABEL or a PARTLABEL
+	if /sbin/blkid | grep "$DEVNAME" | grep "LABEL"; then
+		# Get the LABEL or PARTLABEL
+		LABEL=`/sbin/blkid | grep "$DEVNAME" | sed -e 's/.*LABEL=\"//' -e 's/\".*//'`
```

Comment:
How about we do this before the if condition and use the LABEL variable to evaluate the if condition?

Reply:
For evaluation of LABEL prior to the if condition you can probably use something like
`/sbin/blkid | grep "$DEVNAME" | grep -o 'LABEL=".*"' | cut -d '"' -f2`